Cartesian_kernel/include/CGAL/Cartesian/function_objects.h

```diff
@@ -1757,8 +1756,21 @@ namespace CartesianKernelFunctors {
 	
 	if ( CGAL_NTS is_zero(h.c()) )  // parallel to z-axis
 	  return Vector_3(FT(0), FT(0), FT(1));
-	
-	return Vector_3(-h.b(), h.a(), FT(0));
+
+        FT a = CGAL::abs(h.a()),
+          b = CGAL::abs(h.b()),
+          c = CGAL::abs(h.c());
+
+        // to avoid badly defined vectors with coordinates all close
+        // to 0 when the plane is almost horizontal, we ignore the
+        // smallest coordinate instead of always ignoring Z
+        if (a <= b && a <= c)
+          return Vector_3(FT(0), -h.c(), h.b());
+
+        if (b <= a && b <= c)
+          return Vector_3(-h.c(), FT(0), h.a());
```

Comment:
To avoid extra filter failures with `Epeck`, if propose:
```C++
     if (CGAL::possibly(a <= b && a <= c))
       return Vector_3(FT(0), -h.c(), h.b());

    ￼ if (CGAL::possibly(b <= a && b <= c))
       return Vector_3(-h.c(), FT(0), h.a());
```
The `CGAL::possibly` function is the identity for `bool`, and for the `Epeck` kernel, when the `Boolean` type is `CGAL::Uncertain<bool>`, the `if` condition will be evaluated to `true` is the inequalities are indeterminate using the interval arithmetic. Anyway, the construction is exact and, as we already know at this point that both `a`, `b`, and `c` are non-zero, any of three solutions will produce a valid base vector.